internal/auth/oidc/repository_managed_group_members.go

```diff
@@ -0,0 +1,214 @@
+package oidc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+)
+
+// SetManagedGroupMemberships will set the managed groups for the given account
+// ID. If mgs is empty, the set of groups the account belongs to will be
+// cleared. It returns the set of managed group IDs.
+//
+// mgs contains the set of managed groups that matched. It must contain the
+// group's version as this is used to ensure consistency.
```

Comment:
Perhaps add: "it's understood that managed groups are eventually consistent and that these msgs represent the point in time we will consist the msgs to."